engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java

```diff
@@ -206,6 +206,34 @@ public JobRecord updateJobRetries(final long jobKey, final int retries) {
     return job;
   }
 
+  @Override
+  public void cleanupTimeoutsWithoutJob() {
+    deadlinesColumnFamily.whileTrue(
+        (key, value) -> {
+          final var jobKey = key.second().inner();
+          final var deadline = key.first().getValue();
+          final var job = jobsColumnFamily.get(jobKey);
+          if (job == null || job.getRecord().getDeadline() != deadline) {
+            deadlinesColumnFamily.deleteExisting(key);
+          }
+          return true;
+        });
+  }
+
+  @Override
+  public void cleanupBackoffsWithoutJobs() {
+    backoffColumnFamily.whileTrue(
+        (key, value) -> {
+          final var jobKey = key.second().inner();
+          final var backoff = key.first().getValue();
+          final var job = jobsColumnFamily.get(jobKey);
+          if (job == null || job.getRecord().getRetryBackoff() != backoff) {
```

Comment:
:question: I'm not entirely sure if this is safe. Could there be a case where the backoff of this job is not in the backoff column family and we would end up deleting all backoffs for this job?